packages/builder/src/components/design/NavigationPanel/ScreenNameModal.svelte

```diff
@@ -0,0 +1,52 @@
+<script>
+  import { ModalContent, Input } from "@budibase/bbui"
+  import sanitizeUrl from "builderStore/store/screenTemplates/utils/sanitizeUrl"
+  import { selectedAccessRole, allScreens } from "builderStore"
+
+  export let modal
+  export let navigationSelectionModal
+  export let screenName
+  export let url
+
+  let routeError
+  let roleId = $selectedAccessRole || "BASIC"
+
+  const routeChanged = event => {
+    if (!event.detail.startsWith("/")) {
+      url = "/" + event.detail
+    }
+    url = sanitizeUrl(url)
+
+    if (routeExists(url, roleId)) {
+      routeError = "This URL is already taken for this access role"
+    } else {
+      routeError = ""
+    }
+  }
+
+  const routeExists = (url, roleId) => {
+    return $allScreens.some(
+      screen =>
+        screen.routing.route.toLowerCase() === url.toLowerCase() &&
+        screen.routing.roleId === roleId
+    )
+  }
+</script>
+
+<ModalContent
+  size="M"
+  title={"Enter Details"}
```

Comment:
We should probably lowercase the "Details" here to be consistent with the other modals.